deploy/kiali/kiali_cr.yaml

```diff
@@ -691,10 +691,23 @@ spec:
 #    ---
 #    port: 20001
 #
+# Defines the public domain where Kiali is being served. This the "domain" part
+# of the URL (usually it's a full-qualified domain name).
+# For example, "kiali.example.org".
+# When empty, Kiali will try to guess this value from HTTP headers.
+#    ---
+#    web_fqdn: ""
+#
 # Defines the context root path for the Kiali console and API endpoints and readiness probes.
 # When providing a context root path that is not "/", do not add a trailing slash.
 # For example, use "/kiali" not "/kiali/".
 # When empty, will default to "/" on OpenShift and "/kiali" on Kubernetes.
 #    ---
 #    web_root: ""
+#
+# Defines the public HTTP schema used to serve Kiali. This can only take
+# two possible values: either "http" or "https".
+# When empty, Kiali will try to guess this value from HTTP headers.
+#    ---
+#    web_schema: ""
```

Comment:
We do not need this setting at all. You already know this (and we do this kind of check already), by looking at the `identity` settings. In other words, there is no need to ask an admin user to set a setting like this explicitly when by virtue of the admin user setting `identity` you already know if kiali will be using http or https. See: https://github.com/kiali/kiali-operator/blob/v1.18.0/deploy/kiali/kiali_cr.yaml#L518-L525

If identity.cert_file and identity.private_key_file are explicitly left empty, then you know the schema is 'http' by definition. If these are set, then you know the schema is 'https' (again, by definition). Giving Kiali an identity means you have an identity cert that will identify the server via https. We use this very knowledge when setting up the http/https endpoint here: https://github.com/kiali/kiali/blob/05d65e5757ba908332b7b7eecd902644a255822b/server/server.go#L66-L73

Reply:
That's partially true.

When you set the identity option, Kiali will serve via https **within** the cluster. But the ingres, loadbalancer, or whatever exposure method you choose, may use another *public-facing* protocol.

You can try in your minikube setup. Kiali should be listening in plain HTTP because there is no identity by default. But if you use the ingress endpoint, you can access Kiali either through HTTP or HTTPS. Try going to either **https**://{your_minikube_ip}/kiali or **http**://{your_minikube_ip}/kiali and both endpoints should respond adequately as plain/secure http respectively. The other case could also be true: you could fill the `identity` configuration and Kiali pod will be serving with HTTPS, but the ingress/loadbalancer/whatever may be exposing Kiali publicly through plain HTTP.

So, I think the `identity` option is unreliable regarding the *public-facing* protocol and that's the reason I added this option. But well... We could try the "identity" route if you think it's OK enough.

Reply:
hmmm. ok. lemme think about that.